### PR TITLE
INT-6757 init container for IQ server

### PR DESCRIPTION
**Pull request by jflinchbaugh**
Provide a way to introduce initContainers in place of pre-start containers, so a user can run custom initializations before the server starts.

JIRA: https://issues.sonatype.org/browse/INT-6757

### CODE REVIEW

File: helm-charts/iqserver/templates/deployment.yaml [hunk 1]
```diff
@@ -42,11 +42,6 @@ spec:
             exec:
               command: {{ .Values.deployment.postStart.command }}
           {{- end }}
-          {{- if .Values.deployment.preStart.command }}
-          preStart:
-            exec:
-              command: {{ .Values.deployment.preStart.command }}
-          {{- end }}
```
Comment (jflinchbaugh) [inline helm-charts/iqserver/templates/deployment.yaml:49]: no more prestart.

File: helm-charts/iqserver/templates/deployment.yaml [hunk 2]
```diff
@@ -57,10 +52,14 @@ spec:
             name: nxiq-pv-log
           - mountPath: /etc/nexus-iq-server
             name: config-volume
-  {{- if .Values.iq.licenseSecret }}
+          {{- if .Values.iq.licenseSecret }}
           - mountPath: /etc/nexus-iq-license
             name: license-volume
-  {{- end }}
+          {{- end }}
```
Comment (jflinchbaugh) [inline helm-charts/iqserver/templates/deployment.yaml:58]: minor reindentation.

File: helm-charts/iqserver/templates/deployment.yaml [hunk 3]
```diff
@@ -57,10 +52,14 @@ spec:
...
+      {{- if .Values.initContainers }}
+      initContainers:
+        {{- toYaml .Values.initContainers | nindent 8 }}
```
Comment (jflinchbaugh) [inline helm-charts/iqserver/templates/deployment.yaml:61]: straight import of whatever the user gives us for initContainers.

File: scripts/templates/nxiq-operator-certified.clusterserviceversion.yaml [hunk 4]
```diff
@@ -290,6 +286,19 @@ spec:
...
+    If you need custom initialization, you can use standard Kubernetes
+    [`initContainers`](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+    to run any series of containers and commands you need.
+
+    ```yaml
+    initContainers:
+      - name: an-init-container
+        image: busybox:1.28
+        command: ['/bin/sh', '-c', 'echo "initializing"']
+    ```
```
Comment (jflinchbaugh) [inline scripts/templates/nxiq-operator-certified.clusterserviceversion.yaml:301]: new documentation with sample.

**Review by eduard-tita — APPROVED** [review]
+1 LGTM

**Review by hectorhuol — APPROVED** [review]
+1  LGTM

**Pull request merged**